docs.feldera.com/docs/pipelines/transactions.md

```diff
@@ -0,0 +1,256 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Efficient Bulk Data Processing using Transactions
+
+:::warning
+Transaction support is an experimental feature and may undergo significant
+changes, including non-backward-compatible modifications, in future releases of
+Feldera.
+:::
+
+Transactions let Feldera pipelines to ingest and process large volumes of data atomically—in
+one logical unit of work rather than piece-by-piece.  Transactions are used to achieve:
+
+* **Efficient backfill**: ingest and process large historical datasets.
+
+* **Atomicity**: process multiple inputs atomically without emitting intermediate results.
+
+## Overview
+
+By default, Feldera pipelines run in **continuous mode**: they ingest a chunk of
+data from input connectors, process it to completion and produce updates to all
+views before processing the next chunk.  This mode is optimal for low-latency
+incremental view maintenance; but less suited for bulk data ingest.
+
+A common scenario is when a pipeline must ingest a large volume of historical data
+accumulated over years before processing new real-time inputs. This is known as
+the **backfill** problem. While backfill can be performed in continuous mode,
+this is likely to lead to two issues:
+
+* **Performance:** computing all intermediate updates can be more expensive than
+  computing the cumulative update. The exact performance difference depends on
+  your SQL queries and can be very significant in some cases.
+
+* **Atomicity:** in continuous mode the pipeline produces a stream of intermediate
+  updates, which often cancel each other out.  Consider for instance a `COUNT(*)`
+  query whose output changes any time new records are ingested.  These updates
+  expose intermediate computation results to downstream systems while placing
+  excessive load on the data sinks.
+
+**Transactions** solve these challenges by allowing input updates to be grouped into
```

Comment:
based on this discussion it follows that transactions are also useful even if data volumes are not high - when the users DO want to have control over which intermediate results are visible.